examples/proto/main.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+
+	"github.com/gogo/gateway"
+	// "github.com/grpc-ecosystem/go-grpc-middleware/validator"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/grpclog"
+
+	"github.com/jaegertracing/jaeger/examples/proto/server"
+	"github.com/jaegertracing/jaeger/model/proto/api_v2"
+)
+
+var (
+	gRPCPort    = flag.Int("grpc-port", 10000, "The gRPC server port")
+	gatewayPort = flag.Int("gateway-port", 11000, "The gRPC-Gateway server port")
+)
+
+var log grpclog.LoggerV2
+
+func init() {
+	log = grpclog.NewLoggerV2(os.Stdout, os.Stderr, os.Stderr)
+	grpclog.SetLoggerV2(log)
+}
+
+// serveOpenAPI serves an OpenAPI UI on /openapi-ui/
+// Adapted from https://github.com/philips/grpc-gateway-example/blob/a269bcb5931ca92be0ceae6130ac27ae89582ecc/cmd/serve.go#L63
+// func serveOpenAPI(mux *http.ServeMux) error {
+// 	mime.AddExtensionType(".svg", "image/svg+xml")
+
+// 	statikFS, err := fs.New()
+// 	if err != nil {
+// 		return err
+// 	}
+
+// 	// Expose files in static on <host>/openapi-ui
+// 	fileServer := http.FileServer(statikFS)
+// 	prefix := "/openapi-ui/"
+// 	mux.Handle(prefix, http.StripPrefix(prefix, fileServer))
+// 	return nil
+// }
+
+// Test: curl http://localhost:11000/api/v1/traces/something
+func main() {
+	flag.Parse()
+	addr := fmt.Sprintf("localhost:%d", *gRPCPort)
+	lis, err := net.Listen("tcp", addr)
+	if err != nil {
+		log.Fatalln("Failed to listen:", err)
+	}
+	s := grpc.NewServer(
+	// grpc.Creds(credentials.NewServerTLSFromCert(&insecure.Cert)),
+	// grpc.UnaryInterceptor(grpc_validator.UnaryServerInterceptor()),
+	// grpc.StreamInterceptor(grpc_validator.StreamServerInterceptor()),
+	)
+	api_v2.RegisterQueryServiceServer(s, server.New())
```

Comment:
Just wanted to mention that it's probably a good idea to have each service on its own port, so that different components can have different constraints. Like, I might want to expose the Query Service to the public internet but not the Collector.